src/assignments/PAUL_OMAKI/assignment03.clj

```
@@ -0,0 +1,195 @@
+(ns assignments.PAUL-OMAKI.assignment03)
+
+;; problem 128
+
+
+(defn card-suit-interpreter
+  "Accepts standard suit letters (D, H, C, S) of playing card notation and outputs a value that can be interpreted by Clojure. Input should be a single character."
+  [char]
+  (let [letter-to-test char]
+    (case letter-to-test
+      \D :diamond    ; If it begins with a D, the suit is diamond, etc.
+      \H :heart
+      \C :club
+      \S :spade
+      "Case not handled yet.")))
```

It would be better to let it throw some sort of exception, if the code path is incorrect we should let it throw and be explicit rather then have some weird text value as a return which could then proliferate elsewhere in the code causing a more confusing error